docs/ilab-model-engine.md

```diff
@@ -0,0 +1,204 @@
+2. ambiguity in "what" we are generating, training etc.
+    - The current structure of `ilab` requires an in depth knowledge of the ordering of commands and background knowledge for how our CLI is built. This is not the best way to grow. Commands that are not intuitive and require a deep dive into the docs just to get started might not engage users as we hope to. having an intuitive structure in which people can see easily from the docstring what each commands does and how they might use them, makes more sense. We are generating data, then training a model, running a server to interact with that model, then finally chatting with the model
+3. some commands like `ilab test` and `ilab list` should be renamed to show their true purpose (`ilab model inference`and `ilab taxonomy diff`)
+4. we will hit a point where we want to add more commands to `ilab` due to expected integration with the backend, model registries etc. This list will grow to a point where this form of top group organization is not maintainable.
+
+In general, moving to this structure allows for more growth when `ilab` launches publicly and encourages broader usage of the tool for all sorts of open source AI models.
+
+Outside the scope of this EP, but still important in the near future, the following structure is preferred:
+
+```console
+ilab
+|
+|_______model
+|       |
+|       |____quantize *
+|       |____convert
+|       |____download
+|       |____train (--convert)
+|       |____top *
+|       |____serve   (-i)
+|       |____chat
+|       |____inference
+|       |____push *
+|       |____rm *
+|       |____ls *
+|       |____tag *
+|       |
+|_______data
+|       |____generate
+|       |____mix *
+|       |
+|_______config
+|       |
+|       |____create *
+|       |____edit *
+|       |____rm *
+|       |
+|_______tax
+|       |
+|       |____diff
+|       |____check
+|       |____download
+```
+
+The starred commands are new, and will be dealt with in another enhancement.
+
+#### Goals for End of May release in CLI
+
+For the next few milestones, it has been identified that the overall structure shift should be in place. We should aim to have the vague hierarchy of commands in place that will exist for the future of `ilab` so that when users become familiar with the CLI, it does not change radically soon after launch.
+
+The primary focus here should be the `ilab model` group. The other ones should not be treated as blockers for launch, but would be nice to have.
+
+Another key change that should be included is a switch to using `/var/lib/models` as a model store and `$HOME/.config/models/ilab.conf` for the new configuration path. Keeping config and models in the CWD does not map well to the importance of these two pieces. This will allow `ilab config` commands to act on a global level per-user. Models will be a system-wide directory as they are with most other applications of this type.
```

Comment:
Do we want global config? The thinking from the start is that a user can create multiple independent project s on their system, similar to OperatorSDK.

Reply:
I think global model store and config makes sense.

At least global model store IMO, but if you have that it makes sense to have a global config pointing to this model store.

The way we use config today is really limited, most users override 90% of the config per ilab run, so having duplicates all around the system doesn't rly make sense anymore.

Reply:
Agree with Charlie - I think we can start with a global config, but we can always enhance later for users to have options to create separate "projects" (there's some existing work for something like that here https://github.com/instructlab/instructlab/pull/1022)

Reply:
I would prefer to remove this from this proposal. Changing the CLI command structure is not dependent on making this configuration management change.

Reply:
There are some thorny details with changing the data store, so agree that ideally its split out into a separate proposal.

Ex:
/var/lib/models is normally not writable, so I assume maybe you were thinking XDG_DATA_HOME/ilab/models (~/.local/share/ilab/models) and XDG_CONFIG_HOME/ilab (~/.config/ilab)

Other questions are what happens if someone has two working areas, do all checkpoints end up in the model directory or do they end up in the workspace.

Also relatedly, huggingface recently abandoned this approach with a user cache directory for model, so something to think about. I agree with the goal that we probably want an easy way to share downloaded models.

Reply:
It's a detail but using `/var/lib/ilab/models` is preferred.